apis/python/src/tiledbsoma/_sparse_nd_array.py

```diff
@@ -348,15 +424,14 @@ def used_shape(self) -> Tuple[Tuple[int, int], ...]:
         Compare this to ``shape`` which returns the available/writable capacity.
         """
         retval = []
-        with tiledb.open(self.uri, ctx=self.context.tiledb_ctx):
-            for i in range(20):
-                lower_key = f"soma_dim_{i}_domain_lower"
-                lower_val = self.metadata.get(lower_key)
-                upper_key = f"soma_dim_{i}_domain_upper"
-                upper_val = self.metadata.get(upper_key)
-                if lower_val is None or upper_val is None:
-                    break
-                retval.append((lower_val, upper_val))
+        for i in range(20):
```

Comment:
not directly related to this change but: where did 20 come from?

Reply:
I put that in a long time ago & didn't comment it well : (

Reply:
Good question. I think this should the number of dimensions in the array and not hardcoded to 20? 

@johnkerl

Reply:
> Good question. I think this should the number of dimensions in the array and not hardcoded to 20?

@nguyenv yes